src/components/subscriptions/edit/EditQuotas.js

```diff
@@ -0,0 +1,203 @@
+import React, { useState } from "react";
+import { useQuery, useMutation, useQueryClient } from "react-query";
+import { Field, FieldArray, Form, Formik } from "formik";
+import { mapPropsToValues } from "./formatters";
+import DEDialog from "components/utils/DEDialog";
+import { Button } from "@material-ui/core";
+import FormTextField from "components/forms/FormTextField";
+
+import {
+    updateUserQuotas,
+    getSubscriptions,
+    SUBSCRIPTIONS_QUERY_KEY,
+} from "serviceFacades/subscriptions";
+import { announce } from "components/announcer/CyVerseAnnouncer";
+import { useTranslation } from "react-i18next";
+
+import ids from "../ids";
+import { nonEmptyField } from "components/utils/validations";
+import ErrorTypographyWithDialog from "components/error/ErrorTypographyWithDialog";
+import buildID from "components/utils/DebugIDUtil";
+
+import Usages from "./Usages";
+import Quotas from "./Quotas";
+
+function EditQuotasDialog(props) {
+    const { open, onClose, parentId, subscription } = props;
+    const [quotasSubmission, setQuotasSubmission] = useState([]);
+    const [selectedUsername, setSelectedUsername] = useState(null);
+    const [updateQuotasError, setUpdateQuotasError] = useState(null);
+    const { t } = useTranslation("subscriptions");
+    const queryClient = useQueryClient();
+    const [selectedSubscription, setSelectedSubscription] = useState(null);
+
+    useQuery({
+        queryKey: [
+            SUBSCRIPTIONS_QUERY_KEY,
+            { parentId: parentId, user: subscription?.user.username },
+        ],
+        queryFn: () =>
+            getSubscriptions({ searchTerm: subscription?.user.username }),
+        enabled: true,
+        onSuccess: (data) => {
+            setSelectedSubscription(data.result.subscriptions[0]);
```

Comment:
I thought this was the same info that also came from the subscription listing, which is already being passed to this component in the `subscription` prop.

So maybe there is no need for this query after all?

If so, then all references to `selectedSubscription` in this component could be replaced with the `subscription` prop instead.

Reply:
Initially I had left this query out, but I found that the component wasn't updating the quota values after an update. At least, it wasn't updating consistently. I spent some time debugging this but wasn't able to resolve it by invalidating the query.